ClientRuntimes/CSharp/ClientRuntime/StringTokenProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Rest
+{
+    /// <summary>
+    /// A simple token provider that always provides a static access token.
+    /// </summary>
+    public sealed class StringTokenProvider : ITokenProvider
+    {
+        private string _accessToken;
+        private string _type;
+
+        /// <summary>
+        /// Create a token provider for bearer tokens that always returns the given 
+        /// access token.
+        /// </summary>
+        /// <param name="accessToken">The access token for this token provider to return.</param>
+        public StringTokenProvider(string accessToken) : this(accessToken, "Bearer")
```

Comment:
The "Bearer" has been used multiple time, suggest to use a pre-defined constants